gpm_core/src/config.rs

```diff
@@ -0,0 +1,121 @@
+use crate::remote::Remote;
+use anyhow::Context;
+use serde::{Deserialize, Serialize};
+use std::env;
+use std::fs;
+use std::path::{Path, PathBuf};
+
+/// Represents the configuration used by the CLI.
+#[derive(Deserialize, Serialize)]
+pub struct Config {
+    pub general: ConfigGeneral,
+}
+
+#[derive(Deserialize, Serialize)]
+pub struct ConfigGeneral {
+    pub remotes: Vec<Remote>,
+}
+
+/// Where the config file is stored depending on the OS:
+/// - `%userprofile%/appdata/local/games-package-manager/config.toml` on Windows
+/// - `${home}/.config/games-package-manager/config.toml` on Linux
+///
+/// _I don't know if this code should have been in `crate::constants` or not. I felt
+/// it fit here as it's used only by the Config code. Feel free to move it if you
+/// don't like the current location_
+static CONFIG_RELATIVE_PATH: &str = "games-package-manager/config.toml";
+
+fn get_config_path() -> Result<PathBuf, std::io::Error> {
+    let error_if_not_found = Err(std::io::Error::new(
+        std::io::ErrorKind::NotFound,
+        "Could not find home directory",
+    ));
+    let absolute_path = dirs::config_dir().map(|p| p.join(CONFIG_RELATIVE_PATH));
+
+    match absolute_path {
+        None => error_if_not_found,
+        Some(path) => Ok(path),
+    }
+}
+
+impl Config {
+    pub fn new(remotes: Vec<Remote>) -> Config {
+        Config {
+            general: ConfigGeneral { remotes: remotes },
+        }
+    }
+
+    /// Loads the config from the disk and returns a `Config` with the data it read.
+    /// it expect to find in a
+    pub fn config_from_disk() -> anyhow::Result<Config> {
+        let config_path = get_config_path().context(format!("Could not find home directory"))?;
+
+        let config_content = fs::read_to_string(config_path).context(format!(
```

Comment:
maybe it is a better idea to try to create it here rather than before the call to the remote commands.

Reply:
do you mean the `write_default_config_if_not_exist` call? I separated them on purpose as i thought a function supposed to read from the disk is not supposed to write to the disk as the name implies.

But i agree people may forget to call it, i don't like having a read_from_disk function writing to the disk. It's confusing. But perhaps i could add add a wrapper function called `get_config()` that would call both functions and return a new config. It would keep things simple and solve the issue.

Reply:
It indeed seem to be a good idea.